src/firmware/app/control/trajectory_planner.h

```diff
@@ -11,293 +13,434 @@ typedef struct FirmwareRobotPathParameters
  * the planner will assume the max acceleration that will remain bellow this limit.
  *
- *     - To ensure the robot is capable of deccelerating to react to any sudden changes in
+ *     - To ensure the robot is capable of decelerating to react to any sudden changes in
  * the path, the trajectory is checked for backwards continuity
  *
  * NOTE: The following function is based on an algorithm described in page 18 of
  * http://www2.informatik.uni-freiburg.de/~lau/students/Sprunk2008.pdf
  *
- * @pre path_parameters.num_segments > 2 and num_segments <=
- * TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS
- * @pre path_parameters.max_allowable_acceleration > 0
- * @pre path_parameters.max_allowable_speed > 0
- * @pre path_parameters.init_speed >= 0
- * @pre path_parameters.final_speed >= 0
- * @pre The trajectory.trajectory_elements[] array is pre-allocated to handle up to the
- * TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS limit
- *
- * @param path_parameters [in] The data structure including important path parameters as
- * defined by the FirmwareRobotPathParameters struct
- * @param trajectory [out] The planned trajectory that follows robot dynamics limitations
- * with the appropriate guarantees and assumptions outlines above
- *
- * @return The outcome of the trajectory generation. Returns OK if trajectory is valid.
+ * @pre path_parameters members have been completely initialized and have values
+ *
+ * @param position_trajectory [out] The trajectory data struct to be modified to contain
+ * the position trajectory
+ *
+ * @param path_parameters [in] The path parameters that define the physical limitations
+ * and profile of the trajectory.
+ *
+ * @return A status indicating whether or not generation was successful
  */
 TrajectoryPlannerGenerationStatus_t
-app_trajectory_planner_generateConstantArcLengthPositionTrajectory(
-    PositionTrajectory_t* trajectory);
+app_trajectory_planner_generateConstantParameterizationPositionTrajectory(
+    PositionTrajectory_t* position_trajectory,
+    FirmwareRobotPathParameters_t path_parameters);
 
 /**
- * Returns a constant interpolation period (time) trajectory based on an input trajectory
+ * Function that modifies an existing speed profile to be backwards continuous. This means
+ * that it is possible to reach every speed node within the limits of deceleration
  *
- * This function is intended to take a variable time trajectory (which is easier to
- * calculate) then interpolate a constant period trajectory. This is valuable for discrete
- * time motion controllers that operate only at constant delta-time intervals
+ * @pre All arrays are pre-allocated up to at least TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS
  *
- * NOTE: This function does not modify the real trajectory. It will output a physically
- * invalid trajectory if the input trajectory is invalid
+ * @param speeds [in/out] The existing speed profile to be modified in-place to be
+ * backwards continuous
```

Comment:
`in/out` should come after `in`, as per our new style guide entry

Reply:
Changed